test/unit/spec/EnrollOnPrem_spec.js

```diff
@@ -99,6 +99,22 @@ function (Q, _, $, OktaAuth, Util, Form, Beacon, Expect, $sandbox,
             Expect.isTextField(test.form.credentialIdField());
           });
         });
+        itp('autopopulates credentialId text field', function () {
+          return setup().then(function (test) {
+            expect(test.form.getCredentialId()).toEqual('test123');
+          });
+        });
+        itp('does not autopopulate credentialId when profile does not exist', function () {
+          delete resAllFactors['response']['_embedded']['factors'][3]['profile'];
```

Comment:
I think it will be a good idea to have a ```setup``` method that takes care of copying and deleting the object we don't need. Something like ```setupWithoutProfile```